### ISSUE TITLE
!bpcmds Fails

### ISSUE THREAD

**Issue by tmr232** [bootleg/ret-sync#8]
In some cases, the `!bpcmds save` commands fails. When it does, I get the following output in IDA, and Windbg hangs.

```
[-] Sync failed to parse json
 {"type":"bps_set","msg":"censored-base64-sensitive-data"
------------------------------------
\[<] bad hdr '}'
[-] Request dropped due to bad header
```

Since this only happens once every few attempts, it is a bit tricky to reproduce (failed to do that on purpose so far) and therefore tricky to debug.

**Comment by tmr232** [comment]
Ok... A little digging about shows that the issue is in the passing of data from the broker to the plugin.

In the broker, we print the command to the buffer at https://github.com/bootleg/ret-sync/blob/master/ext_ida%2Fbroker.py#L73.

In my tests, the plugin read the command before the flush operation at https://github.com/bootleg/ret-sync/blob/master/ext_ida%2FSyncPlugin.py#L746, so it first reads most of the buffer, and then the terminating `}`.

**Comment by bootleg** [comment]
Closing this one as the issue described should have been fixed with commit 55a3fd62708b3b9e24d3b44ed160343a2d5126c0. Root cause was poor network communication handling code.

Thank you for reporting it.
